Tests/Marketplace/reinstall_packs.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import os
+import sys
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from pathlib import Path
+import demisto_client
+from Tests.Marketplace.configure_and_install_packs import search_and_install_packs_and_their_dependencies
+from Tests.Marketplace.search_and_uninstall_pack import uninstall_pack
+from Tests.configure_and_test_integration_instances import CloudBuild, get_custom_user_agent
+from Tests.scripts.utils import logging_wrapper as logging
+from Tests.scripts.utils.log_util import install_logging
+
+
+def options_handler() -> argparse.Namespace:
+    """
+    Returns: options parsed from input arguments.
+    """
+    parser = argparse.ArgumentParser(description='Utility for testing re-installation XSIAM packs.')
+    parser.add_argument('--cloud_machine', help='cloud machine to use, if it is cloud build.')
+    parser.add_argument('--cloud_servers_path', help='Path to secret cloud server metadata file.')
+    parser.add_argument('--cloud_servers_api_keys', help='Path to the file with cloud Servers api keys.')
+    parser.add_argument('--non-removable-packs', help='List of packs that cant be removed.')
+    parser.add_argument('--build-number', help='CI job number where the instances were created', required=True)
+    parser.add_argument('--packs_to_reinstall', help='List of packs to check its re-installation.', required=True)
+
+    options = parser.parse_args()
+
+    return options
+
+
+def reinstall_packs(options: argparse.Namespace, cloud_machine: str) -> bool:
+    """
+
+    """
+    success = True
+    api_key, _, base_url, xdr_auth_id = CloudBuild.get_cloud_configuration(cloud_machine,
+                                                                           options.cloud_servers_path,
+                                                                           options.cloud_servers_api_keys)
+
+    client = demisto_client.configure(base_url=base_url,
+                                      verify_ssl=False,
+                                      api_key=api_key,
+                                      auth_id=xdr_auth_id)
+
+    client.api_client.user_agent = get_custom_user_agent(options.build_number)
+    host = client.api_client.configuration.host.replace('https://api-', 'https://')  # disable-secrets-detection
+
+    logging.debug(f'Setting user agent on client to: {client.api_client.user_agent}')
+
+    non_removable_packs = options.non_removable_packs.split(',')
+    packs_to_reinstall_path = Path(options.packs_to_reinstall)
+    packs_to_reinstall = packs_to_reinstall_path.read_text().split('\n')
+
+    for pack in packs_to_reinstall:
```

Comment:
let's add an option to decide if one-by-one or all at once (you already have `uninstall_packs` so it's not a lot of work)
hopefully someday we will be able to use it to significantly speed up the build :)

Reply:
If I install it in batches and not one by one, it will be difficult to understand which pack failed. 
Regarding uninstallation, I can change it but If I use the function `uninstall_packs`, I also need to use: `wait_for_uninstallation_to_complete` which will not work in this case. (Should be changed as well). 
This will require more effort to update and check it. 
What do you think?

Reply:
If the effort is too high, then ok.